xdcs-agent/src/main/python/xdcs/kernel.py

```diff
@@ -0,0 +1,211 @@
+from enum import Enum
+
+import numpy as np
+import pyopencl as cl
+import xdcs.computing.OpenClKernelRunner as Compiler
+
+from xdcs.app import xdcs
+from xdcs.cmd.object_repository import RetrieveObjectsCmd
+from xdcs_api.agent_execution_pb2 import KernelConfig
+
+
+class KernelArgument:
+    class Direction(Enum):
+        IN = 'IN'
+        OUT = 'OUT'
+        IN_OUT = 'IN_OUT'
+
+    class Type(Enum):
+        SIMPLE = 'SIMPLE'
+        POINTER = 'POINTER'
+
+    _name: str
+    _value: str
+    _position: int
+    _direction: Direction
+    _type: Type
+
+    def __init__(self, name: str, value: str, position: int, direction: Direction, arg_type: Type):
+        self._name = name
+        self._value = value
```

Comment:
to value to nie powinno być czasem `_object_id`? bo wszędzie używasz terminu "argument" albo "value" co może sugerować, że to jest wartość

Reply:
poprawione na czytelniejsze rozwiązanie